### ISSUE TITLE
BDD based test suite

### ISSUE THREAD

**Issue by ajtucker** [GSS-Cogs/databaker#3]
Before we start refactoring this codebase or upgrading any dependencies, we need some test coverage so that we can check that we're not breaking existing use.

By now we have a large number of examples of use and expectations that we can check. The API is well documented along with tutorials, so we should be able to use this documentation along with our usage and some example spreadsheets to help create a suite of Gherkin feature files that, once implemented, can act as a test suite to help us catch any breaking changes we make.

- [x] Set up [Behave](https://behave.readthedocs.io/en/latest/index.html).
- [x] Create feature files covering the uses in the tutorials.
- [x] Add example spreadsheets as fixtures.
- [x] Implement steps.
- [x] Set up CI #6 
- [x] Fix failing tests
- [x] Have pull request approved

**Comment by ajtucker** [comment]
Now that the tests are running on Jenkins, quite a few test failures seem to be about not being able to read XLSX files, which I'm sure we've figured out somewhere.

https://ci.floop.org.uk/job/GSS_data/job/databaker/job/bdd-tests/11/cucumber-html-reports/report-feature_994238831.html

**Comment by CharlesRendle** [comment]
FIxed the XLSX issues by changing all fixture datasets to XLS. I'm not sure if this was the best solution in case this alters the formatting to the point where the databaker functions are now incorrect but all features in which this was an issue now pass.
Problem seemed to be caused by a patch to xlrd in which support was removed for anything other than XLS files: https://stackoverflow.com/questions/65254535/xlrd-biffh-xlrderror-excel-xlsx-file-not-supported

Attribute error in spaceprefix feature cause and solution found: https://github.com/beetbox/beets/issues/2986
I have amended this line to overrides.py but given the date of this issue (2018) I don't know why this hasn't caused us an issue already or why it has only come into effect now.

Will tick the final box on the ticket
